notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/ConnectionString.java

```diff
@@ -14,6 +14,18 @@
     private String mSharedAccessKey;
 
     ConnectionString(String endpoint, String sharedAccessKeyName, String sharedAccessKey) {
+        if (isNullOrWhiteSpace(endpoint)) {
+            throw new IllegalArgumentException(ENDPOINT_KEY);

```

Comment:
I just checked the IllegalArgumentException documentation, and I think we should probably have more details than this in the exception message. This feedback applies to all three calls here :)